server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -295,9 +302,11 @@ private Response performSegmentUpdate(
           .build();
     }
 
-    final ImmutableDruidDataSource dataSource = getDataSource(dataSourceName);
-    if (dataSource == null) {
-      return logAndCreateDataSourceNotFoundResponse(dataSourceName);
+    if (validateDatasourceIsQueryable) {
```

Comment:
Perhaps include a comment to the effect of:

```suggestion
// We can skip the markUnused operation when there are no used segments for this datasource.
// However, for the markUsed operation, we should still continue as we want to mark a portion of the unused segments as used.
    if (dataSource == null && !isMarkUsed) {
```

Reply:
changed this logic